llama_index/llms/portkey.py

```diff
@@ -0,0 +1,366 @@
+"""
+    Portkey intergation with Llama_index for enchanced monitoring
+"""
+from typing import Any, Optional, Sequence, Dict, Union, List
+
+from rubeus import (
+    Rubeus,
+    LLMBase,
+    RubeusModes,
+    RubeusModesLiteral,
+    ProviderTypes,
+    ProviderTypesLiteral,
+    RubeusCacheType,
+    RubeusCacheLiteral,
+    Message,
+    RubeusResponse,
+)
```

Comment:
If you want to keep these imports for typing, take a look at the marvin extractor for an example as well
https://github.com/jerryjliu/llama_index/blob/main/llama_index/node_parser/extractors/marvin_metadata_extractor.py

Reply:
Made the changes by moving them under the TYPE_CHECKING condition.